### ISSUE TITLE
Submission failing to submit

### ISSUE THREAD

**Issue by StuartMcD** [solspace/craft3-freeform#1364]
### What happened?

Have a form which has one field, email address. Notification template takes the submitted email address and parses where logic looks for the email address elsewhere in the system and either delivers a "yay or "nay" 

Template code:

```
{{ freeform.form("commRequest", {
	attributes : {
		class: "form w-full",
		id: "comms",
	},
	values: {
		email: displayName ?? null,
	},
	buttons: {
		attributes: {
			submit: {class: 'bg-theme text-theme-accent px-5 py-2 rounded font-normal mt-4 md:mt-0 md:ml-auto shadow-md hover:shadow hover:translate-y-px' },
		},
	},
}).render() }}
```

On submission, I now get a 500 response outlined below, the live version works as per usual as it's the older version still.

### Errors and Stack Trace (if available)

```shell
Exception 'TypeError' with message 'Solspace\Freeform\Services\SubmissionsService::getSubmissionById(): Argument #1 ($id) must be of type int, null given, called in /var/www/html/vendor/solspace/craft-freeform/packages/plugin/src/Jobs/SendNotificationsJob.php on line 39' 

in /var/www/html/vendor/solspace/craft-freeform/packages/plugin/src/Services/SubmissionsService.php:55

Stack trace:
#0 /var/www/html/vendor/solspace/craft-freeform/packages/plugin/src/Jobs/SendNotificationsJob.php(39): Solspace\Freeform\Services\SubmissionsService->getSubmissionById(NULL)
#1 /var/www/html/vendor/solspace/craft-freeform/packages/plugin/src/Jobs/FreeformQueueHandler.php(30): Solspace\Freeform\Jobs\SendNotificationsJob->execute(Object(craft\queue\Queue))
#2 /var/www/html/vendor/solspace/craft-freeform/packages/plugin/src/Bundles/Notifications/SendListeners/EmailRecipientNotifications.php(77): Solspace\Freeform\Jobs\FreeformQueueHandler->executeNotificationJob(Object(Solspace\Freeform\Jobs\SendNotificationsJob))
#3 [internal function]: Solspace\Freeform\Bundles\Notifications\SendListeners\EmailRecipientNotifications->sendToRecipients(Object(Solspace\Freeform\Events\Forms\SendNotificationsEvent))
#4 /var/www/html/vendor/yiisoft/yii2/base/Event.php(312): call_user_func(Array, Object(Solspace\Freeform\Events\Forms\SendNotificationsEvent))
#5 /var/www/html/vendor/solspace/craft-freeform/packages/plugin/src/Bundles/Notifications/NotificationsBundle.php(61): yii\base\Event::trigger('Solspace\\Freefo...', 'send-notificati...', Object(Solspace\Freeform\Events\Forms\SendNotificationsEvent))
#6 [internal function]: Solspace\Freeform\Bundles\Notifications\NotificationsBundle->sendNotifications(Object(Solspace\Freeform\Events\Forms\SendNotificationsEvent))
#7 /var/www/html/vendor/yiisoft/yii2/base/Event.php(312): call_user_func(Array, Object(Solspace\Freeform\Events\Submissions\ProcessSubmissionEvent))
#8 /var/www/html/vendor/solspace/craft-freeform/packages/plugin/src/Services/SubmissionsService.php(200): yii\base\Event::trigger('Solspace\\Freefo...', 'process-submiss...', Object(Solspace\Freeform\Events\Submissions\ProcessSubmissionEvent))
#9 /var/www/html/vendor/solspace/craft-freeform/packages/plugin/src/Services/SubmissionsService.php(170): Solspace\Freeform\Services\SubmissionsService->postProcessSubmission(Object(Solspace\Freeform\Form\Types\Regular), Object(Solspace\Freeform\Elements\Submission))
#10 /var/www/html/vendor/solspace/craft-freeform/packages/plugin/src/controllers/SubmitController.php(46): Solspace\Freeform\Services\SubmissionsService->handleSubmission(Object(Solspace\Freeform\Form\Types\Regular))
#11 [internal function]: Solspace\Freeform\controllers\SubmitController->actionIndex()
#12 /var/www/html/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#13 /var/www/html/vendor/yiisoft/yii2/base/Controller.php(178): yii\base\InlineAction->runWithParams(Array)
#14 /var/www/html/vendor/yiisoft/yii2/base/Module.php(552): yii\base\Controller->runAction('', Array)
#15 /var/www/html/vendor/craftcms/cms/src/web/Application.php(340): yii\base\Module->runAction('freeform/submit', Array)
#16 /var/www/html/vendor/craftcms/cms/src/web/Application.php(641): craft\web\Application->runAction('freeform/submit', Array)
#17 /var/www/html/vendor/craftcms/cms/src/web/Application.php(302): craft\web\Application->_processActionRequest(Object(craft\web\Request))
#18 /var/www/html/vendor/yiisoft/yii2/base/Application.php(384): craft\web\Application->handleRequest(Object(craft\web\Request))
#19 /var/www/html/public_html/index.php(22): yii\base\Application->run()
#20 {main}l/vendor/solspace/craft-freeform/packages/plugin/src/Jobs/SendNotificationsJob.php on line 39'`
```

### How can we reproduce this?

1. Update FF from 5.3.31 to 5.4.1
2. Update cms from 4.9.7 to 4.10.2
3. Use ajax form
4. No integrations
5. Hangs on submit

### Freeform Edition

Pro

### Freeform Version

5.4.1

### Craft Version

5.4.1

### When did this issue start?

Unsure

### Previous Freeform Version

5.3.31

**Comment by StuartMcD** [comment]
I believe this may have been an issue at my end to do with date format or reCaptcha - c;losing to confirm!

**Comment by StuartMcD** [comment]
Further to this issue, it's only occurring on forms that do NOT Store Submitted Data

**Comment by kjmartens** [comment]
I'm very sorry about this @StuartMcD.

Thanks for tracking it down to the Store Submitted Data setting. I am able to duplicate this as well. We will get a fix out for it shortly. 🙂

**Comment by kjmartens** [comment]
This is now resolved in Freeform 5.5.1. 🙂

**Comment by mtwalsh** [comment]
Not sure this is definitely fixed as I'm still hitting this issue with forms that do not store submitted data on `5.5.2`.

**Comment by kjmartens** [comment]
Sorry @mtwalsh. I can duplicate this as well. We were testing a variety of situations with the latest fix and somehow this was missed or there was a regression along the way.

Very sorry about this... we'll get a fix available shortly.

**Comment by kjmartens** [comment]
Ok @mtwalsh and @StuartMcD,

This should now finally be resolved in Freeform 5.5.3. 🙂

**Comment by mtwalsh** [comment]
Thanks for the prompt fix on this @kjmartens , looks good from here 👍
